src/datreant/data/pydata.py

```diff
@@ -52,4 +54,10 @@ def get_data(self, key, **kwargs):
                 the selected data
         """
         with self.read():
-            return pickle.load(self.handle)
+            # load in bytes with python3 to ensure successful read EVERYTIME.
+            # Using the ascii encoding it can happen that python3 can read a
+            # pickle written with python 2.
+            try:
+                return pickle.load(self.handle, encoding='bytes')
```

Comment:
I'm not sure about this anymore. I know it can help to unpack things. But I would have rather use it as a last resort before giving up.